### PR TITLE
Updates to JS component usage section

### PR DESCRIPTION
**Pull request by adidahiya**

### CODE REVIEW

File: packages/core/src/components/_index.scss [hunk 1]
```diff
@@ -70,7 +63,8 @@ Styleguide components.usage.npm
 Vanilla JS APIs
 
 JS Components are built using React, but that does not limit their usage to just React applications.
-You can render any component in any JavaScript application with `ReactDOM.render`.
+You can render any component in any JavaScript application with `ReactDOM.render`. Think of it like
+using a JQuery plugin.
```
Comment (giladgray) [inline packages/core/src/components/_index.scss]: if you're going to make this analogy then you're going to have to back it up.

`ReactDOM.render(<Plugin {...options} />, selector)` ~~ `$(selector).plugin(options)`

File: packages/core/src/components/_index.scss [hunk 2]
```diff
@@ -30,17 +24,18 @@ Styleguide components.usage
...
+Install the core package with an NPM client like `npm` or `yarn`, pulling in all relevant dependencies:
 
 ```sh
 npm install --save @blueprintjs/core
 ```
 
-Finally, you'll need to include the global type definitions for React and ReactDOM in your project using
-[Typings][typings] or [TSD][tsd]. Blueprint's own typings, `blueprint.d.ts`, are distributed in the NPM module and
-should be resolved automatically by the [compiler][tscompiler].
+If you are using TypeScript, you'll need to include ambient type definitions for React and ReactDOM in your
```
Comment (giladgray) [inline packages/core/src/components/_index.scss]: would be cool to have a complete list of the minimal set of typings necessary. that should be considered part of installing blueprint.

Comment (giladgray) [inline packages/core/src/components/_index.scss]: i know it's not, you haven't mentioned CSSTransitionGroup 😄

Comment (adidahiya) [inline packages/core/src/components/_index.scss]: what do you mean? you get that with the react / react dom typings...

File: packages/core/src/components/_index.scss (hunk 1 above)
Comment (adidahiya) [inline packages/core/src/components/_index.scss]: will address later, trying to get simple fixes in now.

**Pull request merged**